File: apps/company-knowledge-ui/src/api/trending.ts
import { apiRequest } from "./client";

export type TrendingQuestion = {
  question: string;
  topic_id?: string | null;
  count: number;
  unique_users?: number;
  last_asked_utc?: string | null;
};

export function fetchTrendingQuestions({ window = "7d", limit = 8 } = {}): Promise<TrendingQuestion[]> {
  return apiRequest<TrendingQuestion[]>(`/api/v1/trending?window=${encodeURIComponent(window)}&limit=${limit}`);
}

File: libs/shared-schemas/python/shared_schemas/trending.py
from datetime import datetime

from pydantic import BaseModel, Field


class TrendingQuestion(BaseModel):
    question: str = Field(min_length=1)
    topic_id: str | None = None
    count: int = Field(ge=1)
    unique_users: int = Field(default=1, ge=1)
    last_asked_utc: datetime | None = None

